src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -221,6 +222,32 @@ public static LocalDate getTargetClassDate(LocalDateTime currentDateTime, LocalT
         }
     }
 
+    /**
+     * Parses a {@code String classTimeRange} into a {@code TimeRange}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code classDateTime} is invalid.
+     */
+    public static TimeRange parseTimeRange(String classTimeRange) throws ParseException {
+        requireNonNull(classTimeRange);
+        String trimmedTimeRange = classTimeRange.trim();
+        System.out.println("trimmedTimeRange is " + trimmedTimeRange);
+        if (TimeRange.isValidTimeRange(trimmedTimeRange)) {
+            LocalTime startTime = parseTime(trimmedTimeRange.substring(0, 4));
+            LocalTime endTime = parseTime(trimmedTimeRange.substring(5, 9));
+            Integer duration = Integer.valueOf(trimmedTimeRange.substring(10));
+            if (!Class.isValidDuration(startTime, endTime)) {
+                throw new ParseException(Class.INVALID_DURATION_ERROR_MESSAGE);
+            }
+            if (!TimeRange.isValidEndTime(startTime, endTime, duration)) {
+                throw new ParseException(TimeRange.INVALID_DURATION_ERROR_MESSAGE);
+            }
+            return new TimeRange(startTime, endTime, duration);
+        } else {
+            throw new ParseException(TimeRange.MESSAGE_CONSTRAINTS);
+        }
```

Comment:
I think that in this case the current implementation makes more sense because `isValidTimeRange` should return `true` or `false` and not throw an exception.

Reply:
This makes sense! Will consider doing something similar for my bug fix later ;)